build.gradle

```diff
@@ -0,0 +1,52 @@
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath "com.android.tools.build:gradle:1.2.3"
+        classpath "com.jakewharton.sdkmanager:gradle-plugin:0.12.0"
+    }
+}
+
+apply plugin: "android-sdk-manager"
+apply plugin: "com.android.application"
+
+android {
+    compileSdkVersion 22
+    buildToolsVersion "22"
+
+    defaultConfig {
+        def versionMajor = 1
+        def versionMinor = 0
+        def versionPatch = 0
+
+        versionName buildVersionName(versionMajor, versionMinor, versionPatch)
+        versionCode buildVersionCode(versionMajor, versionMinor, versionPatch)
+
+        minSdkVersion 16
+        targetSdkVersion 22
+    }
+}
+
+def buildVersionName(versionMajor, versionMinor, versionPatch) {
+    return "${versionMajor}.${versionMinor}.${versionPatch}"
+}
+
+def buildVersionCode(versionMajor, versionMinor, versionPatch) {
+    return versionMajor * 10 ** 2 + versionMinor * 10 ** 1 + versionPatch * 10 ** 0
+}
+
+dependencies {
+    repositories {
+        mavenCentral()
+    }
+
+    compile "com.android.support:support-annotations:22.1.1"
+    compile "com.jakewharton:butterknife:6.1.0"
+    compile "com.jakewharton.timber:timber:3.1.0"
+    compile "com.squareup.dagger:dagger:1.2.2"
+    compile "org.jboss.aerogear:aerogear-android-pipe:2.1.0"
```

Comment:
Do you mean Butter Knife or Dagger? Butter Knife is not a dependency injection, it is just an automatic listeners generation.